pkg/resources/server.go

```diff
@@ -118,18 +122,46 @@ func (rs *resourceServer) NotifyRegistrationStatus(ctx context.Context,
 func (rs *resourceServer) Allocate(ctx context.Context, rqt *pluginapi.AllocateRequest) (*pluginapi.AllocateResponse, error) {
 	glog.Infof("Allocate() called with %+v", rqt)
 	resp := new(pluginapi.AllocateResponse)
+
+	err := rs.cdiCache.Refresh()
```

Comment:
why do we  need to refresh here ?  (when cdi is not enabled we dont, but i also dont see why we need to refresh when CDI is enabled)

Reply:
done